components/_buttons/buttons.scss

```diff
@@ -252,99 +290,282 @@
     &:not(:disabled):not(.disabled).active,
     &.show > .btn-gray.dropdown-toggle {
       background: $danger-red-1;
-      box-shadow: 0 0 0 0.25rem $rgba-5;
+      box-shadow: 0 0 0 4px $rgba-5;
       color: $white;
     }
 
     &:focus,
     &.focus {
-      box-shadow: 0 0 0 0.25rem $rgba-5;
+      box-shadow: 0 0 0 4px $rgba-5;
+    }
+  }
+
+  // Floating Button
+  &--floating {
+    align-items: center;
+    padding: 0px;
+    @include msds-font('body-1');
+    color: $clear-blue;
+    font-weight: 400;
+    text-transform: unset;
+    .msds-icon {
+      width: 13.33px;
+      height: 13.33px;
+      padding: 9.34px;
+      margin-right: 20px;
+      transform: rotate(90deg);
+      border: 1px solid $clear-blue;
+      border-radius: 50%;
+    }
+    &:hover {
+      text-decoration: underline;
+      padding: 0px;
+      .msds-icon {
+        color: $white;
+        background: $clear-blue-gradient;
+        box-shadow: inset 0 -1px 0 0 #0175a6, 0 2px 12px 0 $rgba-1;
+      }
+    }
+
+    &:active {
+      text-decoration: underline;
+      .msds-icon {
+        color: $white;
+        background: $clear-blue-gradient;
+        box-shadow: 0 0 0 4px rgba(0, 153, 218, 0.5);
+      }
+    }
+
+    &:focus {
+      outline: none;
+
+      .msds-icon {
+        box-shadow: 0 0 0 4px rgba(0, 153, 218, 0.5);
+      }
+    }
+    &:disabled {
+      @include floating-light-theme-disabled();
+    }
+  }
+
+  &--floating-sm {
```

Comment:
IMO oppinion this is a problem, We are now adding modifiers to modifiers. We are probably extending the msds-btn to much here. We should stick to the clean code principles and extract this into its own component.

Reply:
I agree, working on it now. Your evil man!!! Refactoring the whole thing :'(